### PR TITLE
Added MbedTLS 3.1.0 project

### PR DESCRIPTION
**Pull request by sergioVaz00**

### CODE REVIEW

**Review by ulisescosti — APPROVED** [review]

File: benchmark-mbedTLS-vector.sh [hunk 1]
```diff
@@ -0,0 +1,272 @@
...
+printf "Wellcome to Codee's interactive demo with MBedTLS\n"
+printf "##################################################\n"
+printf "Seven steps:\n"
+printf "  1. Generate the compile_commads.json and build the neccesary files for the analysis\n"
+printf "  2. Codee's screening report for the whole suite\n"
+printf "  3. Codee's screening report for the most important algorightms\n"
+printf "  4. Vectorize the code with Codee's pwdirectives tool\n"
+printf "  5. Build the vectorized version\n"
+printf "  6. Verify Speed Up of the vectorized version\n"
+printf "  7. Verify correctnes of the vectorized version\n"
```
Comment (ulisescosti) [inline benchmark-mbedTLS-vector.sh]:
```suggestion
printf "  7. Verify the correctness of the vectorized version\n"
```

File: benchmark-mbedTLS-vector.sh [hunk 2]
```diff
@@ -0,0 +1,272 @@
...
+printf "##################################################\n"
+printf "Wellcome to Codee's interactive demo with MBedTLS\n"
+printf "##################################################\n"
+printf "Seven steps:\n"
+printf "  1. Generate the compile_commads.json and build the neccesary files for the analysis\n"
+printf "  2. Codee's screening report for the whole suite\n"
+printf "  3. Codee's screening report for the most important algorightms\n"
+printf "  4. Vectorize the code with Codee's pwdirectives tool\n"
+printf "  5. Build the vectorized version\n"
+printf "  6. Verify Speed Up of the vectorized version\n"
```
Comment (ulisescosti) [inline benchmark-mbedTLS-vector.sh]:
```suggestion
printf "  6. Verify the speedup of the vectorized version\n"
```

File: benchmark-mbedTLS-vector.sh [hunk 3]
```diff
@@ -0,0 +1,272 @@
...
+./ \
+-G Ninja
+
+cmake --build buildVec
+
+#===============================================================================
+printf "\nStep 5 done.\n"
+printf "=============================================================\n\n"
+printf "##################################################\n"
+printf "6/7. Verifying Speed Up of the vectorized version ...\n"
```
Comment (ulisescosti) [inline benchmark-mbedTLS-vector.sh]:
```suggestion
printf "6/7. Verifying the speedup of the vectorized version ...\n"
```

File: benchmark-mbedTLS-vector.sh [hunk 4]
```diff
@@ -0,0 +1,272 @@
...
+printf "original done\n"
+
+buildVec/programs/test/benchmark sha256
+printf "vectorized done\n"
+
+#===============================================================================
+printf "\nStep 6 done.\n"
+printf "=============================================================\n\n"
+printf "##################################################\n"
+printf "7/7. Verifying correctnes of the vectorized version ...\n"
```
Comment (ulisescosti) [inline benchmark-mbedTLS-vector.sh]:
```suggestion
printf "7/7. Verifying the correctness of the vectorized version ...\n"
```

**Review by ulisescosti — APPROVED** [review]

File: README.md [hunk 5]
```diff
@@ -32,21 +43,23 @@ The top-level folder of the repository is organized as follows. Different code e
...
+- VMS: Virtual Molecule Screening pure_c.
 
-Inside each one of them, the `serial` subfolder corresponds to the original sequential (ie. non-parallelized) code. Additionally, one or more subfolders corresponding to performance-optimized versions are provided. The `pwa-omp-multi` folder corresponds to an OpenMP multi-threaded version generated by Parallelware Analyzer, which requires an OpenMP compiler like `gcc` or `clang` properly installed. The `pwa-acc-offload` folder corresponds to an OpenACC GPU offloading version generated by Parallelware Analyzer, which requires and OpenACC compiler like `nvc` properly installed. 
+Inside each one of  (except MbedTLS), the `serial` subfolder corresponds to the original sequential (ie. non-parallelized) code. Additionally, one or more subfolders corresponding to performance-optimized versions are provided. The `pwa-omp-multi` folder corresponds to an OpenMP multi-threaded version generated by Codee, which requires an OpenMP compiler like `gcc` or `clang` properly installed. The `pwa-acc-offload` folder corresponds to an OpenACC GPU offloading version generated by Codee, which requires and OpenACC compiler like `nvc` properly installed.
```
Comment (ibogosavljevic) [inline README.md]: If I remember correctly @manuelarenaz  requested that all the tests in the performance-demos follow the same format. The user would then just run `make` in the root directory to measure before and after time.

File: README.md [hunk 6]
```diff
@@ -68,3 +81,53 @@ MATMUL          5.49    1.04    5.28x   81.07%
...
+```
+...
+##################################################
+7/7. Verifying the speedup of the vectorized version ...
+##################################################
+Press enter to continue
+
+
+++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
+aes_xts algorithm
```
Comment (ibogosavljevic) [inline README.md:34]: Another algorithm that experiences speed improvements with this is AES-CBC. @sergioVaz00, please confirm!

Comment (sergioVaz00) [reply README.md:34]: Yes, even if only one of the loops is vectorized, a small speed improvement is obtained:
original:
  ARIA-CBC-128             :     156038 KiB/s,         18 cycles/byte
  ARIA-CBC-192             :     137425 KiB/s,         20 cycles/byte
  ARIA-CBC-256             :     122271 KiB/s,         22 cycles/byte

vectorized:
  ARIA-CBC-128             :     165943 KiB/s,         17 cycles/byte
  ARIA-CBC-192             :     145935 KiB/s,         19 cycles/byte
  ARIA-CBC-256             :     129778 KiB/s,         21 cycles/byte

Do you want me to add it so you can make the new benchmark?

**Pull request merged**